packages/protocol/contracts/team/TimelockTokenPool.sol

```diff
@@ -168,18 +198,28 @@ contract TimelockTokenPool is EssentialContract {
 
     function _withdraw(address recipient, address to) private {
         Recipient storage r = recipients[recipient];
+
+        uint256 price;
         uint128 amount;
 
         uint256 rGrantsLength = r.grants.length;
         for (uint128 i; i < rGrantsLength; ++i) {
-            amount += _getAmountUnlocked(r.grants[i]);
+            uint128 perGrantAmount;
+
+            perGrantAmount = _getAmountUnlocked(r.grants[i]);
```

Comment:
it will be great to merge these 3 lines into the _getAmountUnlocked and rename it as _getAmountAndCostUnlocked

Reply:
Here is a logical error: `r.amountWithdrawn` is the total amount withdrawn for all grants assigned to this address, previously this is ok, but now each grant has a different strike price, you have to calculated the cost per grant with its own `amountWithdrawn`. The Recipient must be modified to remove `amountWithdrawn` and put `amountWithdrawn` inside each grant`. How tests didn't capture this?

Reply:
Because it is a loop per grant. And it multiply the per grant amount with per grant price, so there is no logical error here.

Reply:
Amount withdrawn is TKO.
I calculated here the cost per grant (in the loop).

amountWithdrawn and amountWithdrawable is per overall tokens not per grant! (You wrote that part of the code).

Who really cares about amountWithdrawn per grant ? If Y portion is withdrawable on grant A, and X portion is withdrawable on grant B, than -> amount withdrawabel is : Y + X

Reply:
Done.
Tests were not catching because i set the strikePrices to 0 every existing tests, and added 2 new, but they did not intermediary steps, just X years after 100% fully unlocks, so this kind of case did not come up. Now i modified all files to have a non-zero strike price.